core/new-gui/src/app/common/service/user/workflow-persist/workflow-persist.service.ts

```diff
@@ -28,6 +28,28 @@ export class WorkflowPersistService {
       .pipe(map(WorkflowPersistService.parseWorkflowInfo));
   }
 
+  /**
+   * creates a workflow and insert it to backend database and returns its information
+   * @param newWorkflowName
+   * @param newWorkflowCreationTime
+   * @param newWorkflowLastModifiedTime
+   * @param newWorkflowContent
+   */
+  public createWorkflow(newWorkflowName: string, newWorkflowCreationTime: number|undefined, newWorkflowLastModifiedTime: number|undefined,
```

Comment:
when creating a new workflow, the name and content should be provided. The name has a default value. Other fields should not be provided.

Reply:
Resolved, will be updated in the next commit